### ISSUE TITLE
[Devops] Increase ease of deployment via setup scripts/buttons from GitHub repo

### ISSUE THREAD

**Issue by thisistimothychen** [bitcamp/hackathon-expo-app#7]

**Comment by thisistimothychen** [comment]
Sync with Joshua Fleming (helped out a lot with Technica Expo App deployment via EC2 + Gunicorn + Nginx)

**Comment by chris-sul** [comment]
### Proposed Solution
I think first to aid both development and deployment we can containerize the the backend. With this, we could completely automate python package installation and gunicorn setup. This would also keep a consistent platform for the app to run on in case any platform dependent issue arose. In addition, depending on how he do customization, we could even have an image that we publish that other hackathons could just pull and use. I suggest we use docker and docker-compose for this. As a side note, using docker-compose we could also have images for both the frontend and backend be created automatically at the same time so someone could have the whole app deployed in one go.

Since we are already in the AWS infrustructure we could also leverage aws ecs which sits on top of ec2 in case we need to do any load balancing depending on how the app works under load. As an added benefit, if we ever wanted to do automated testing of the app having it in the container would make that a lot easier.

SSL certificates are probably still going to have to be manually installed, but we could script out the nginx and docker installation/configuration once that is done

### Timeline
- By 2/1 have the app build into a Docker container using docker-compose (with gunicorn)
- By 2/15 have the nginx/docker installation/configuration scripts completed
- By 3/1 have the simpler deployment instructions completed
- Test

**Comment by thisistimothychen** [comment]
@codemastermd this is amazing 😍 sounds like you know what you're doing!
CC-ing @nyaculak (registration/mobile backend team lead) who is also working on deployment-related stuff for (also Docker I believe) - might be good/easy to keep these two similar in terms of what infrastructure we're going to be using

**Comment by nyaculak** [comment]
@codemastermd this is essentially the same as my plan for the registration app.

I have 3 different services that will all be run in Docker containers (mongodb, node/express backend, nextjs frontend).

I was also doing some research and I think we could also deploy nginx through Docker as well, but this is obviously optional and we could add it later.

I recommend starting by making a simple docker-compose that runs your database and the backend. If your client is hosted on github pages right now, it shouldn't need to change too much. Later, you can add the client to the docker-compose as well.

**Comment by thisistimothychen** [comment]
@nyaculak sweet, thanks for the insight! :)
Also, @codemastermd @nyaculak thoughts on just hosting our own MongoDB instance on that EC2 instance? Can that be incorporated into the docker stuff and automatically deployed?

**Comment by thisistimothychen** [comment]
Goal (Cloudformation, ability to deploy to AWS using a script): Friday, Feb. 22
Afterwards: documentation

**Comment by chris-sul** [comment]
Update - 
Having some trouble getting the CloudFormation stuff to work but otherwise the containerization / nginx stuff is working.

Going to split this up into multiple PR's, one in the next day or two for the nginx and docker stuff (2/25). Then another one once I can get the rest of the cloud formation stuff to work (3/1). After that documentation which should not take too long.

**Comment by chris-sul** [comment]
Deployment works, going to double check documentation
